server/controllers/users.js

```diff
@@ -324,16 +325,39 @@ module.exports = (app) => {
      * Show.
      */
     show: (req, res, next) => {
-      if (req.remoteUser.id === req.user.id) {
+
+      var userData = req.user.show;
+
+      if (req.remoteUser && req.remoteUser.id === req.user.id) {
         req.user.getStripeAccount()
           .then((account) => {
-            var response = _.extend({}, req.user.info, { stripeAccount: account });
-
+            var response = _.extend(userData, req.user.info, { stripeAccount: account });
             res.send(response);
           })
           .catch(next);
-      } else {
-        res.send(req.user.show);
+      } else if (req.query.profile) {
+        var groups;
+        req.user
+        .getGroups()
+        .then(results => {
```

Comment:
can simplify to `.then(results => results.map(g => Object.assign(g.minimal, { role: g.UserGroup.role })))`